### DISCUSSION TITLE
Why not all authorization requirements included in ForbiddenResponseOperationFilter?

### DISCUSSION THREAD

**Discussion by xperiandri** [ASP-NET-MVC-Boilerplate/Framework#14] [Q&A]
Why only [these](https://github.com/ASP-NET-Core-Boilerplate/Framework/blob/master/Source/MVC6/Boilerplate.AspNetCore.Swagger/OperationFilters/ForbiddenResponseOperationFilter.cs#L44) requirements are included?

**Comment by RehanSaeed** [comment]
Forbidden means you have access to the site but you are not allowed to access the resource within the site. Are there other you think should be included?

**Comment by xperiandri** [comment]
Why not just check for any `IAuthorizationRequirement`?

**Comment by RehanSaeed** [comment]
You don't want to include `DenyAnonymousAuthorizationRequirement` because that produces a 401 (See [code](https://github.com/ASP-NET-Core-Boilerplate/Framework/blob/master/Source/MVC6/Boilerplate.AspNetCore.Swagger/OperationFilters/UnauthorizedResponseOperationFilter.cs)).

Also we can't be sure if any custom requirements a developer creates should create a 401 or 403, so I'm very explicit here about types that should create a 403.

**Comment by xperiandri** [comment]
I thought that requirement must produce 403 only

**Comment by RehanSaeed** [comment]
`DenyAnonymousAuthorizationRequirement` means you must be logged in which translates to a 401. Everything else is a 403 i.e. you don't have access to that resource.
